public/scripts/openai.js

```diff
@@ -299,97 +321,100 @@ function formatWorldInfo(value) {
 }
 
 async function prepareOpenAIMessages(name2, storyString, worldInfoBefore, worldInfoAfter, extensionPrompt, bias, type, quietPrompt) {
-    const isImpersonate = type == "impersonate";
-    let this_max_context = oai_settings.openai_max_context;
-    let enhance_definitions_prompt = "";
-    let nsfw_toggle_prompt = oai_settings.nsfw_toggle ? oai_settings.nsfw_prompt : oai_settings.nsfw_avoidance_prompt;
-
-    // Experimental but kinda works
-    if (oai_settings.enhance_definitions) {
-        enhance_definitions_prompt = "If you have more knowledge of " + name2 + ", add to the character's lore and personality to enhance them but keep the Character Sheet's definitions absolute.";
+    const chatCompletion = promptManager.getChatCompletion();
+
+    // Prepare messages
+    const enhanceDefinitionMessage = chatCompletion.makeSystemMessage(substituteParams('If you have more knowledge of {{char}}, add to the character\'s lore and personality to enhance them but keep the Character Sheet\'s definitions absolute.'));
```

Comment:
It is possible. I moved the enhanced definitions in this commit: https://github.com/SillyTavern/SillyTavern/pull/427/commits/9e2ac951a089e5fd296c98408d7c915d9cd85f2b

As for group nudges and start new chat, it would be possible as well. It makes sense, too, but it also kinda clutters the list from the get-go.